components/data-bridge/org.wso2.carbon.databridge.agent/src/test/java/org/wso2/carbon/databridge/agent/test/DataPublisherTestCase.java

```diff
@@ -0,0 +1,262 @@
+            DataEndpointAgentConfigurationException, TransportException, DataEndpointException,
+            DataEndpointConfigurationException, MalformedStreamDefinitionException, DataBridgeException,
+            StreamDefinitionStoreException, IOException {
+        startServer(9219, 9319);
+        AgentHolder.setConfigPath(DataPublisherTestUtil.getDataAgentConfigPath(agentConfigFileName));
+        String hostName = DataPublisherTestUtil.LOCAL_HOST;
+
+        DataPublisher dataPublisher = new DataPublisher("Binary", "tcp://" + hostName + ":9219",
+                "ssl://" + hostName + ":9319", "admin", "admin");
+
+        String streamID = DataBridgeCommonsUtils.generateStreamId(STREAM_NAME, VERSION);
+        Object[] metaData = new Object[]{"127.0.0.1"};
+        Object[] correlationData = null;
+        Object[] payLoad = new Object[]{"WSO2", 123.4, 2, 12.4, 1.3};
+        long timeout = 1000;
+
+        int numberOfEventsSent = 1000;
+        for (int i = 0; i < numberOfEventsSent; i++) {
+            dataPublisher.tryPublish(streamID, metaData, correlationData, payLoad, timeout);
+        }
+        try {
+            Thread.sleep(5000);
+        } catch (InterruptedException e) {
+        }
+        dataPublisher.shutdown();
+        AssertJUnit.assertEquals(numberOfEventsSent, testServer.getNumberOfEventsReceived());
+        testServer.resetReceivedEvents();
+        testServer.stop();
+    }
+
+    @Test
+    public void nonBlockingPublishTestOverLoadWithArbitaryElementsOfEvent() throws DataEndpointAuthenticationException,
+            DataEndpointAgentConfigurationException, TransportException, DataEndpointException,
+            DataEndpointConfigurationException, MalformedStreamDefinitionException, DataBridgeException,
+            StreamDefinitionStoreException, IOException {
+        startServer(9629, 9729);
+        AgentHolder.setConfigPath(DataPublisherTestUtil.getDataAgentConfigPath(agentConfigFileName));
+        String hostName = DataPublisherTestUtil.LOCAL_HOST;
+
+        DataPublisher dataPublisher = new DataPublisher("Binary", "tcp://" + hostName + ":9651",
+                "ssl://" + hostName + ":9751", "admin", "admin");
+
+        String streamID = DataBridgeCommonsUtils.generateStreamId(STREAM_NAME, VERSION);
+        Object[] metaData = new Object[]{"127.0.0.1"};
+        Object[] correlationData = null;
+        Object[] payLoad = new Object[]{"WSO2", 123.4, 2, 12.4, 1.3};
+        Map<String, String> arbitrary = new HashMap<String, String>();
+        arbitrary.put("test", "testValue");
+        arbitrary.put("test1", "test123");
+
+
+        int numberOfEventsSent = 1000;
+        for (int i = 0; i < numberOfEventsSent; i++) {
+            dataPublisher.tryPublish(streamID, metaData, correlationData, payLoad, arbitrary);
+        }
+        try {
+            Thread.sleep(5000);
+        } catch (InterruptedException e) {
+        }
+        dataPublisher.shutdown();
```

Comment:
Where are the asserts? Applies to all other testcases as well

Reply:
Fixed with commit [#1ab4f1d9e9b2c37d7266d546f4c09b775b4af81a](https://github.com/pcnfernando/carbon-analytics-common/commit/1ab4f1d9e9b2c37d7266d546f4c09b775b4af81a)